PoESkillTree.Engine.GameModel/PassiveTree/PassiveNodeDefinition.cs

```diff
@@ -10,6 +11,42 @@ public class PassiveNodeDefinition : IDefinition<ushort>
             => (Id, Type, Name, IsAscendancyNode, CostsPassivePoint, Position, Modifiers) =
                 (id, type, name, isAscendancyNode, costsPassivePoint, position, modifiers);
 
+        private PassiveNodeDefinition(JsonPassiveNode passiveNode)
+            : this(passiveNode.Id,
+                  passiveNode.PassiveNodeType,
+                  passiveNode.Name,
+                  passiveNode.IsAscendancyNode,
+                  !passiveNode.IsRootNode && !passiveNode.IsAscendancyNode && !passiveNode.IsMultipleChoiceOption,
+                  new NodePosition(passiveNode.Position.X, passiveNode.Position.Y),
+                  passiveNode.StatDescriptions)
+        { }
+
+        public static PassiveNodeDefinition Convert(JsonPassiveNode passiveNode, float zoomLevel = 1f)
+        {
+            static float? SetZoomLevel(JsonPassiveTreePosition? position, float? _zoomLevel)
```

Comment:
I'm not really a fan of this solution:

1. I don't expect a `Convert` function to modify state. It's mostly fine because the original is restored, but this might lead to threading issues. E.g. it might be a good idea to change `GameModel.GameData` to take in a `JsonPassiveTree` or `Task<JsonPassiveTree>` instead of the node definitions after the app has been updated to use this. Should the app tree construction run in parallel to the game data loading, that could lead to issues.
2. `PassiveNodeDefinition` should always be based on the same zoom level. Otherwise things like `PassiveTreeDefinitionExtensions.GetNodesInRadius` (at least the overload that takes in a JewelRadius) will no longer work.

You could add a `PositionAtZoomLevel(float zoomLevel)` function to `JsonPassiveTreePosition`, which would be overloaded instead of and used by the `Position` property. The `PassiveNodeDefinition` constructor could then call that function with the zoom level for which `JewelRadius.GetRadius` is correct.

Reply:
Not sure why I didn't think of that, easy solution haha. It makes the convert method a lot nicer as well. The ZoomLevel should always be at 1f because that was what `JewelRadius.GetRadius()` is based on. The application will need to use the `JewelRadius.GetRadius(float zoomLevel)` in order to properly draw the circle. 

The draw code is going to need a bit of an update when switching to this. It should make it a ton cleaner.